src/BlazorWebView/src/Maui/Windows/BlazorWebView.Windows.cs

```diff
@@ -24,7 +24,10 @@ protected override void OnPropertyChanged(string? propertyName = null)
 
 		private void Window_Destroying(object? sender, EventArgs e)
 		{
-			// see: https://github.com/microsoft/microsoft-ui-xaml/issues/6872
+			// See: https://github.com/dotnet/maui/issues/7277
+			Handler?.DisconnectHandler();
```

Comment:
@PureWeen will this cause issues at all?

Reply:
@MackinnonBuck why is this specifically needed on Windows and not other platforms?

Reply:
The original issue that this PR is solving was reported for Windows. On Android and iOS, the `Window.Destroying` event does not get invoked when the app gets closed (#6993), so it wouldn't make a difference anyway. That said, we could make this line of code platform-independent, but that might introduce a risk that it doesn't work as expected on Android and iOS if/when #6993 gets fixed. Do you think it's worth investigating another way to make this work on other platforms?

Reply:
Each platform has a version of `Unloaded` if you want to tie into those? 
Android => ViewDetachedFromWindow
iOS => override WillMoveToWindow

It doesn't cover the window destroying case but it makes the behavior symmetrical for when a view is removed from the visual hierarchy.

Reply:
@PureWeen this PR already adds a handler for the `Unloaded` event to invoke `DisconnectHandler()`. Just so I understand, is your suggestion to use the platform-specific APIs instead?

Reply:
I guess the general question is: what is the recommended pattern for components/controls to dispose dependencies when they go away? Like say a media control has to dispose some native player component, or Blazor needs to dispose Razor objects, or an image control needs to dispose some file watcher component.

Reply:
Aw sorry @MackinnonBuck I was thinking this was subscribing to `Unloaded` on the platform view. So in that case I think you'd want to subscribe to "loaded" inside `HandlerChanging` when a handler is first assigned. Unless a BlazorWebView is not reusable? Because there are cases when a user will keep around the reference to a `ContentPage` or possibly this `BlazorWebView`

> I guess the general question is: what is the recommended pattern for components/controls to dispose dependencies when they go away? Like say a media control has to dispose some native player component, or Blazor needs to dispose Razor objects, or an image control needs to dispose some file watcher component.

Currently we leave that up to the user based on what they want to do. For example, with a media control we can't really assume when a user wants to dispose of the platform components with a media control. 

Let's say a user wants to move a media control from one layout to another (switching to full screen, absolute, or fixed). In XF this was really hard to do because we would automatically dispose of the entire platform hierarchy. Same question I would say applies here as well. If a BlazorWebView is removed from the visual hierarchy does that mean you always want to dispose the platform components? For example, on Android if you have more than 3 tabs Android will remove your view from the visual hierarchy (so unloaded will fire) if it's not a tab that's to the left or right of the current tab. Is that ok here? Is this going to mess users up when using a BlazorWebView and a TabbedPage with more then 3 tabs? 

In a perfect world we we'd just rely on the GC so when the VirtualView/PlatformView are no longer referenced anywhere they'd just get cleaned up and finalized but I don't think this is going to work in 100 percent of the cases.

Adding some additional hooks for more information about when to disconnect/dispose will probably need to be added but at this point we leave the components life cycle up to the discretion of the component and hopefully the events that we do surface helps give enough information to make that decision. 

All that being said :-) Given the original issue maybe this can't really be fixed until we fix the `window` destroy life cycle events?

Reply:
Hmm, this is indeed more nuanced than I had considered. Is there any logical sense of "this things is definitely garbage and I definitely want to destroy everything", aside from waiting for GC, which is unpredictable? Or is this just not logically possible and it's truly up to each component to define a _heuristic_ (which I define to mean "provably incorrect logic that approximates truth") and work off of that?

Reply:
Ignoring a possible failure of imagination, not really. 

A user might hold a reference to a Page for the entire life time of an app. For example, now with `Shell` since you can register pages as singleton a lot of people are going to register those pages as singletons which means every single control on that page will live for the entire life of the application. 

I think what we'll need to look at as we gather use cases are places where we can make good enough guesses and then let people influence those guesses.  For example, with Shell maybe we should always disconnect the handler if they've registered something as transient? I think one area here that would delight users is if they could scope to a page/navpage/window and then once that container is disposed that will naturally dispose of the children. We can't really do that currently with the limitations of MS.Ext.DI but just spit balling here :-) 

>  "this things is definitely garbage and I definitely want to destroy everything",
TLDR; not really but we should figure out a way to let users tap into life cycle points better.

Reply:
https://github.com/dotnet/maui/issues/7381

Reply:
@PureWeen Since the primary use case of BlazorWebView will be MAUI Blazor apps where the WebView is basically the entire app, do you think a decent direction here would be to run cleanup logic when the PlatformView gets unloaded (which I think is what you were suggesting earlier)? It seems the alternative is to leave things the way they are where nothing ever gets disposed, which probably isn't ideal.

Reply:
@MackinnonBuck 

If the underlying handler is removed and then the same BlazorWebView instance gains a new handler later down the road. Does the BlazorWebView retain enough information about current state that the user would basically just start over from where they were? or would they go back to the beginning as if it was loaded for the first time?

Reply:
@PureWeen how about implementing a reference counter? If I'm not wrong `Comet` has this implemented. If that's a simple implementation we can add this to .NET MAUI and call the `DisconnectHandler` when we don't have anymore references at all.

Reply:
@PureWeen `BlazorWebView` retains hardly any state at all (it's managed by the underlying platform-specific WebView), so the user would basically be starting over from the beginning.

Reply:
> the primary use case of BlazorWebView will be MAUI Blazor apps where the WebView is basically the entire app

That might be the most common pattern, but certainly not the only one. I think _optimizing_ for the most common pattern can make sense, but we have to be careful to not cause significant harm to less-common scenarios (I'm not saying any suggestion here violates that, but I want to make sure we don't end up there). This stuff is very hard for me to think through 😄

Reply:
@Eilon, if we compare MAUI Blazor with Blazor WebAssembly for a minute, we don't call `Dispose()` for Blazor WebAssembly components when, for example, the browser window is closed. If someone needs some logic to run when the application window is closing, couldn't you argue that the resource being disposed exists in a context broader than the BlazorWebView, so therefore the disposal logic should be handled from that broader context (in MAUI lifecycle events, for example)? For example, @PureWeen brought up offline that the issue faced in #7277 could be fixed by running the disposal logic from within a MAUI context. This obviously has the downside that if you want to share Razor components between supported Blazor platforms, you would potentially need multiple versions of the component that manage state/disposal logic differently. But this comes back to what I said earlier about there not really being an equivalent case in Blazor WebAssembly, so maybe that concern isn't valid?

Reply:
The conclusion we reached offline is that we're going to hold off on having `BlazorWebView` determine when to call `DisconnectHandler()` for now. The existing behavior is more acceptable than initially thought, so we feel comfortable revisiting this topic later when the general concept of disposal in MAUI is better understood.